tests/installation/addon_products_sle.pm

```diff
@@ -17,23 +17,49 @@ use testapi;
 use utils 'addon_license';
 use version_utils 'sle_version_at_least';
 use qam 'advance_installer_window';
-use registration '%SLE15_DEFAULT_MODULES';
+use registration qw(%SLE15_DEFAULT_MODULES rename_scc_addons);
 
 sub handle_all_packages_medium {
     assert_screen 'addon-products-all_packages';
     send_key 'alt-s';
-    # We could reuse SCC_MODULES or another list. For now just hardcode what
-    # corresponds to the selected SLE15 product because the "all packages"
-    # addon medium and feature of installer is only available for SLE >= 15
-    # anyway
+
+    # For SLE installation / upgrade with the all-packages media, user has
+    # to select the required extensions / modules manually
     my @addons = split(/,/, $SLE15_DEFAULT_MODULES{get_required_var('SLE_PRODUCT')});
     push @addons, 'desktop' if check_var('DESKTOP', 'gnome') && !grep(/^desktop$/, @addons);
+
+    # The legacy module is required if upgrade from previous version
+    push @addons, 'legacy' if get_var('UPGRADE') && !grep(/^legacy$/, @addons);
```

Comment:
hm, why is that? I don't think it's the idea of the legacy module to be always required

Reply:
The legacy module turns out to be always required in upgrading, which is noted in https://bugzilla.suse.com/show_bug.cgi?id=1066338#c5
It says:
"If you update from "pain" SLES12 without modules added to SLES15, you need minimal the basesystem module, the server application module, the desktop application module and the legacy module. Everything else will not work."

I think such info should be included in release notes or migration guide (if we have such doc).

Reply:
I guess for the test it is fine if you just add "(bsc#1066338)" to the comment in L31. That would help to find the comment by tkukuk and therefore the reasoning from development point of view.